packages/general/src/log/LogDestination.ts

```diff
@@ -0,0 +1,113 @@
+/**
+ * @license
+ * Copyright 2022-2025 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ImplementationError } from "#MatterError.js";
+import { Console } from "./Console.js";
+import { Diagnostic } from "./Diagnostic.js";
+import { LogFormat } from "./LogFormat.js";
+import { LogLevel } from "./LogLevel.js";
+
+/**
+ * An endpoint for log messages.
+ *
+ * You can replace the default destination or add additional destinations to control how log messages are stored.
+ */
+export interface LogDestination {
+    /**
+     * The name of the destination, used for configuration.
+     */
+    name: string;
+
+    /**
+     * The maximum level to log if not set explicitly.
+     */
+    level: LogLevel;
+
+    /**
+     * The maximum level to log for specific facility names.
+     */
+    facilityLevels: Record<string, LogLevel>;
+
+    /**
+     * Contextual information used to optimize log output.
+     */
+    context?: Diagnostic.Context;
+
+    /**
+     * Add an entry to the log.
+     *
+     * The default implementation formats using {@link format} and writes using {@link write}.
+     */
+    add(message: Diagnostic.Message): void;
+
+    /**
+     * Format a log message.
+     *
+     * The default {@link add} implementation uses this to format message for {@link write}.
+     */
+    format(message: Diagnostic.Message): string;
+
+    /**
+     * Write a formatted message to the log.
+     *
+     * The default {@link add} implementation uses this to write messages after formatting.
+     */
+    write(text: string, message: Diagnostic.Message): void;
+}
+
+/**
+ * Create a new {@link LogDestination}.
+ */
+export function LogDestination<T extends Partial<LogDestination>>(
+    config?: T,
+): T extends { add: () => unknown } ? LogDestination : LogDestination {
+    return {
+        ...LogDestination.defaults,
+        ...config,
+    };
+}
+
+export namespace LogDestination {
+    /**
+     * Defaults for {@link LogDestination} fields.
+     */
+    export const defaults: LogDestination = {
+        name: "default",
+
+        level: LogLevel.DEBUG,
```

Comment:
did we not had INFO as default before?

Reply:
No, if you provided an invalid log level to `LogLevel` previously it would select INFO but I made that throw instead.  The actual levels set in `Boot.init` (for default destinations) and in `addLogger` (for tertiary destinations) always defaulted to DEBUG unless explicitly overridden.

That said, we should probably change to INFO as the default soon.